addons/account/models/account_move.py

```diff
@@ -4219,10 +4241,17 @@ def write(self, vals):
         if account_to_write and account_to_write.deprecated:
             raise UserError(_('You cannot use a deprecated account.'))
 
+        inalterable_fields = set(self._get_integrity_hash_fields()).union({'inalterable_hash', 'secure_sequence_number'})
+        violated_fields = ", ".join([
+            self._fields[violated_field].string
+            for violated_field
+            in set(vals).intersection(inalterable_fields)
+        ])
+        if any(line.move_id.inalterable_hash and violated_fields for line in self):
+            hashed_move_names = "\n".join(set(line.move_id.name for line in self if line.move_id.inalterable_hash))
+            raise UserError(_("You cannot edit the following fields: %s.\nThe following entries are already hashed:\n%s", violated_fields, hashed_move_names))
```

Comment:
You should also probably separate technical things and display things.
I think that this is a lot more readable/logical
```suggestion
        hashed_moves = self.move_id.filtered('inalterable_hash')
        violated_fields = set(vals) & inalterable_fields
        if hashed_moves and violated_fields:
            raise UserError(_(
                "You cannot edit the following fields: %s.\n"
                "The following entries are already hashed:\n%s", 
                ', '.join(f['string'] for f in self.fields_get(violated_fields)), 
                '\n'.join(hashed_moves.mapped('name')),
            ))
```